src/effects/VST3/VST3Instance.cpp

```diff
@@ -0,0 +1,194 @@
+{
+   if(mWrapper->Initialize(settings, sampleRate, Steinberg::Vst::kRealtime, mProcessingBlockSize))
+   {
+      mInitialDelay = mWrapper->GetLatencySamples();
+      return true;
+   }
+   return false;
+}
+
+size_t VST3Instance::RealtimeProcess(size_t group, EffectSettings& settings, const float* const* inBuf,
+   float* const* outBuf, size_t numSamples)
+{
+   if(group == 0)
+      return mWrapper->Process(inBuf, outBuf, numSamples);
+   return 0;
+}
+
+bool VST3Instance::RealtimeProcessEnd(EffectSettings& settings) noexcept
+{
+   return true;
+}
+
+bool VST3Instance::RealtimeProcessStart(EffectSettings& settings)
+{
+   mWrapper->ConsumeChanges(settings);
+   return true;
+}
+
+bool VST3Instance::RealtimeResume()
+{
+   mWrapper->ResumeProcessing();
+   return true;
+}
+
+bool VST3Instance::RealtimeSuspend()
+{
+   mWrapper->SuspendProcessing();
+   return true;
+}
+
+sampleCount VST3Instance::GetLatency(const EffectSettings& settings, double sampleRate) const
+{
+   if(mUseLatency)
+      return mInitialDelay;
+   return { 0u };
+}
+
+bool VST3Instance::ProcessFinalize() noexcept
+{
+   return GuardedCall<bool>([&]
+   {
+      mWrapper->Finalize();
+      return true;
+   });
+}
+
+bool VST3Instance::ProcessInitialize(EffectSettings &settings, double sampleRate, ChannelNames chanMap)
+{
+   if(mWrapper->Initialize(settings, sampleRate, Steinberg::Vst::kRealtime, mProcessingBlockSize))
+   {
+      mInitialDelay = mWrapper->GetLatencySamples();
+      return true;
+   }
+   return false;;
+}
+
+size_t VST3Instance::GetBlockSize() const
+{
+   return mProcessingBlockSize;
+}
+
+size_t VST3Instance::SetBlockSize(size_t maxBlockSize)
+{
+   mProcessingBlockSize = 
+      static_cast<Steinberg::int32>(std::min(maxBlockSize, mUserBlockSize));
+   return mProcessingBlockSize;
+}
+
+size_t VST3Instance::ProcessBlock(EffectSettings& settings, const float* const* inBlock, float* const* outBlock,
+   size_t blockLen)
+{
+   return mWrapper->Process(inBlock, outBlock, blockLen);
+}
+
+VST3Wrapper& VST3Instance::GetWrapper()
+{
+   return *mWrapper;
+}
+
+unsigned VST3Instance::GetAudioInCount() const
+{
+   //setupProcessing should be called first
+   return CountChannels(
+      mWrapper->mEffectComponent,
+      Steinberg::Vst::kAudio,
+      Steinberg::Vst::kInput,
+      Steinberg::Vst::kMain
+   );
+}
+
+unsigned VST3Instance::GetAudioOutCount() const
+{
+   //setupProcessing should be called first
+   return CountChannels(
+      mWrapper->mEffectComponent,
+      Steinberg::Vst::kAudio,
+      Steinberg::Vst::kOutput,
+      Steinberg::Vst::kMain
+   );
+}
+
+void VST3Instance::ReloadUserOptions()
+{
+   // Reinitialize configuration settings
+   int userBlockSize;
+   GetConfig(mProcessor, PluginSettings::Shared, wxT("Options"),
+      wxT("BufferSize"), userBlockSize, 8192);
+   mUserBlockSize = std::max( 1, userBlockSize );
+   GetConfig(mProcessor, PluginSettings::Shared, wxT("Options"),
+      wxT("UseLatency"), mUseLatency, true);
```

Comment:
There is a long-distance coincidence of repeated string literals here and in `VST3OptionsDialog.cpp`.  It would be nice to clean that up, but it's not necessary now